cpp/test/writer/tsfile_writer_test.cc

```diff
@@ -0,0 +1,208 @@
+                                  common::TSEncoding::PLAIN,
+                                  common::CompressionType::UNCOMPRESSED));
+            writer.register_timeseries(device_name, measure_name,
+                                       common::TSDataType::INT64,
+                                       common::TSEncoding::PLAIN,
+                                       common::CompressionType::UNCOMPRESSED);
+        }
+    }
+
+    for (int i = 0; i < device_num; i++) {
+        std::string device_name = "test_device" + std::to_string(i);
+        for (int j = 0; j < measurement_num; j++) {
+            int max_rows = 100;
+            Tablet tablet(device_name, &schema_vec[i], max_rows);
+            tablet.init();
+            for (int row = 0; row < max_rows; row++) {
+                tablet.set_timestamp(row, 16225600 + row);
+            }
+            for (int row = 0; row < max_rows; row++) {
+                tablet.set_value(row, j, row);
+            }
+            ASSERT_EQ(writer.write_tablet(tablet), E_OK);
+        }
+    }
+
+    ASSERT_EQ(writer.flush(), E_OK);
+    ASSERT_EQ(writer.close(), E_OK);
+}
+
+TEST_F(TsFileWriterTest, WriteMultipleTabletsDouble) {
+    TsFileWriter writer;
+    writer.init(file_);
+
+    const int device_num = 50;
+    const int measurement_num = 50;
+    std::vector<MeasurementSchema> schema_vec[50];
+
+    for (int i = 0; i < device_num; i++) {
+        std::string device_name = "test_device" + std::to_string(i);
+        for (int j = 0; j < measurement_num; j++) {
+            std::string measure_name = "measurement" + std::to_string(j);
+            schema_vec[i].push_back(
+                MeasurementSchema(measure_name, common::TSDataType::DOUBLE,
+                                  common::TSEncoding::PLAIN,
+                                  common::CompressionType::UNCOMPRESSED));
+            writer.register_timeseries(device_name, measure_name,
+                                       common::TSDataType::DOUBLE,
+                                       common::TSEncoding::PLAIN,
+                                       common::CompressionType::UNCOMPRESSED);
+        }
+    }
+
+    for (int i = 0; i < device_num; i++) {
+        std::string device_name = "test_device" + std::to_string(i);
+        for (int j = 0; j < measurement_num; j++) {
+            int max_rows = 200;
+            Tablet tablet(device_name, &schema_vec[i], max_rows);
+            tablet.init();
+            for (int row = 0; row < max_rows; row++) {
+                tablet.set_timestamp(row, 16225600 + row);
+            }
+            for (int row = 0; row < max_rows; row++) {
+                tablet.set_value(row, j, (double)row + 1.0);
+            }
+            ASSERT_EQ(writer.write_tablet(tablet), E_OK);
+        }
+    }
+
+    ASSERT_EQ(writer.flush(), E_OK);
+    ASSERT_EQ(writer.close(), E_OK);
+}
+/*
+TEST_F(TsFileWriterTest, FlushWithoutWriteAfterRegisterTS) {
```

Comment:
Why is this commented?

Reply:
I keep it for uncommited/unmerged bugifx: _Flushing without writing after registering a timeseries will cause a core dump_,  and I left a note below.